docs: Allow AWS IAM SigV4 for SDS AuthN

bcelenza opened the pull request:
Description: Per #8042, update documentation to clarify authentication required for SDS connections, and add notes about credential types in use today.
Risk Level: Low
Testing: Previewed content layout using RST parser.
Docs Changes: See description.
Release Notes: N/A

htuch commented on docs/root/configuration/security/secret.rst:
```diff
@@ -15,7 +15,10 @@ Upstream clusters are handled in a similar way, if a cluster client certificate
 If a static cluster is using SDS, and it needs to define a SDS cluster (unless Google gRPC is used which doesn't need a cluster), the SDS cluster has to be defined before the static clusters using it.
 
-The connection between Envoy proxy and SDS server has to be secure. One option is to run the SDS server on the same host and use Unix Domain Socket for the connection. Otherwise it requires mTLS between the proxy and SDS server. In this case, the client certificates for the SDS connection must be statically configured.
+The connection between Envoy proxy and SDS server has to be secure. One option is to run the SDS server on the same host and use Unix Domain Socket for the connection. Otherwise it requires valid authentication between the proxy and SDS server using gRPC channel credentials, call credentials, or both. Credential types in use today are:
```
Are we confusing authentication and confidentiality here? There's two aspects:
1. We need to ensure that there is always server TLS configured, i.e. an upstream TLS context for Envoy gRPC/REST, and not using someting like grpc::InsecureChannelCredentials() if it's the Google gRPC client.
2. We need to authenticate. This should be over a secure channel, see (1), but this is where client certs or SigV4 come into play.
The docs read as if we can solve (2) with (1) and don't cover Envoy gRPC/REST.

bcelenza replied:
Agree with both points, and will reword. :)

bcelenza commented:
Hey @htuch, want to make sure I've addressed your feedback appropriately. Happy to field suggestions for wording changes if I still haven't quite nailed it.
